### PR TITLE
Added ChargeService command

### LINKED ISSUE #131: [PE-D] Service command does not work?
Likewise for service command, I do not know whether the command is invalid or that the paramters I am inputting is incorrect.

-------------
Labels: `severity.High` `type.FunctionalityBug`
original: FeliciaTay/ped#13

### LINKED ISSUE #149: [PE-D] service command not working
The `service` command does not seem to be working. 

-------------
Labels: `severity.High` `type.FunctionalityBug`
original: haveaqiupill/ped#8

### LINKED ISSUE #160: [PE-D] A feedback should be provided when I key in a command with the wrong format.
-------------
Labels: `severity.Medium` `type.FeatureFlaw`
original: Yuanyu97/ped#4

### LINKED ISSUE #167: [PE-D] The service command mentioned in the UG is being recognised as an invalid command (and hence not working).
-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: muhd97/ped#8

### PR DESCRIPTION
**Pull request by hmtey**
finished, moving on to fetchbill

### CODE REVIEW

File: docs/UserGuide.adoc [hunk 1]
```diff
@@ -156,17 +156,15 @@ Shows the bill for guest with ID G1231231X's stay up till present moment.
 === Charging a service : `service`
 
 Adds a service to the guest's tab. +
-Format: `service n/NAME  p/PRICE [d/DESCRIPTION]...`
+Format: `service i/G1231231X rn/ROOM_NUMBER d/DESCRIPTION c/COST`
```
Comment (dawidjaja) [inline docs/UserGuide.adoc]: That should be i/ID right?

File: src/main/java/seedu/address/logic/commands/ServiceCommand.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+            + PREFIX_ROOMNUMBER + "ROOM_NUMBER "
+            + PREFIX_DESCRIPTION + "DESCRIPTION "
+            + PREFIX_COST + "COST\n"
+            + "Example: " + COMMAND_WORD + " "
+            + PREFIX_ID + "G1231231X "
+            + PREFIX_ROOMNUMBER + "101 "
+            + PREFIX_DESCRIPTION + "massage "
+            + PREFIX_COST + "20.00";
+
+    public static final String MESSAGE_SUCCESS = "Added %1 to the bill of guest %2s for room %3s.";
```
Comment (dawidjaja) [inline src/main/java/seedu/address/logic/commands/ServiceCommand.java]: Does %1 works? I think we need to specify the type. %1s in this case if it's a string.

File: src/main/java/seedu/address/model/ModelManager.java [hunk 3]
```diff
@@ -299,14 +301,26 @@ public void addTier(Tier tier, ArrayList<String> roomNums) {
...
+    public void addService(PersonId personId, RoomId roomId, Service service) {
+        requireAllNonNull(personId, roomId, service);
+        ArrayList<Bill> bills = addressBook.findPersonWithId(personId).get().getBills();
+
+        for (Bill bill : bills) {
+            if (bill.getRoomId().equals(roomId)) {
+                bill.addChargeable(service);
+            }
+        }
+    }
```
Comment (dawidjaja) [inline src/main/java/seedu/address/model/ModelManager.java]: Do you think it's better to put this in Person class? So here, just check whether the person exists.

File: src/main/java/seedu/address/model/hotel/bill/Cost.java [hunk 4]
```diff
@@ -21,7 +21,7 @@
      */
     public Cost(String cost) {
         requireNonNull(cost);
-        checkArgument(isValidCost(cost), MESSAGE_CONSTRAINTS);
+        checkArgument(isValidCost(cost), cost);
```
Comment (dawidjaja) [inline src/main/java/seedu/address/model/hotel/bill/Cost.java]: The second parameters should be the error message

**Review by dawidjaja — DISMISSED** [review]

File: src/main/java/seedu/address/model/hotel/bill/Cost.java (hunk 4 above)
Comment (youaremysky99) [reply src/main/java/seedu/address/model/hotel/bill/Cost.java]: it seems that you haven't resolved this.

Comment (hmtey) [reply src/main/java/seedu/address/model/hotel/bill/Cost.java]: Oh this is outdated, I've resolved it when i fixed the bug for setrcost!

File: src/main/java/seedu/address/model/hotel/person/Person.java [hunk 5]
```diff
@@ -85,6 +87,30 @@ public PersonId getPersonId() {
...
+        for (Bill bill : bills) {
+            if (bill.getRoomId().equals(roomId)) {
+                bill.addChargeable(service);
+                billExists = true;
+                break;
+            }
+        }
+
+        if (billExists == false) {
+            Bill newBill = new Bill(roomId);
```
Comment (youaremysky99) [inline src/main/java/seedu/address/model/hotel/person/Person.java:109]: So now, a person has a list of bills?

Comment (hmtey) [reply src/main/java/seedu/address/model/hotel/person/Person.java:109]: yepp

File: src/main/java/seedu/address/logic/commands/ChargeServiceCommand.java [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+            + PREFIX_SERVICEID + "SERVICE_ID\n"
+            + "Example: " + COMMAND_WORD + " "
+            + PREFIX_ID + "G1231231X "
+            + PREFIX_ROOMNUMBER + "101 "
+            + PREFIX_SERVICEID + "WC";
+
+    public static final String MESSAGE_SUCCESS = "Added service (ID: %1$s) to the bill of "
+            + "guest (ID: %2$s) for room %3$s.";
+    public static final String MESSAGE_GUEST_NONEXISTENT = "Guest (ID: %1$s) does not exist in the system.";
+    public static final String MESSAGE_ROOM_NONEXISTENT = "Room %1s does not exist in the system.";
```
Comment (dawidjaja) [inline src/main/java/seedu/address/logic/commands/ChargeServiceCommand.java]:
```suggestion
    public static final String MESSAGE_ROOM_NONEXISTENT = "Room %1$s does not exist in the system.";
```

Comment (dawidjaja) [reply src/main/java/seedu/address/logic/commands/ChargeServiceCommand.java]: I believe you want to take the argument index right?

File: src/main/java/seedu/address/model/Hotel.java [hunk 7]
```diff
@@ -115,6 +118,19 @@ public boolean hasBooking(Booking booking) {
         return false;
     }
 
+    /**
+     * Checks if {@code person} has booked {@code room}.
+     */
+    public boolean hasGuestBooked(Person person, Room room) {
+        for (Booking b : bookingList) {
```
Comment (dawidjaja) [inline src/main/java/seedu/address/model/Hotel.java:125]: I'm pretty sure bookingList contains future reservation as well. Have you check for this case? I think we can create a method to check whether it's a reservation, or a stay. Clarify with me if this is not clear.

**Review by dawidjaja — CHANGES_REQUESTED** [review]
Hi, could you fix these?

**Review by youaremysky99 — APPROVED** [review]

File: src/main/java/seedu/address/model/Hotel.java [hunk 8]
```diff
@@ -115,6 +119,19 @@ public boolean hasBooking(Booking booking) {
         return false;
     }
 
+    /**
+     * Checks if {@code person} is checked into {@code room}.
+     */
+    public boolean isGuestCheckedIn(Person person, Room room) {
+        for (Booking b : bookingList) {
+            if (b instanceof Stay && b.getPayee().equals(person) && b.getRoom().equals(room)) {
```
Comment (dawidjaja) [inline src/main/java/seedu/address/model/Hotel.java]:
```suggestion
            if (b.getPayee().equals(person) && b.getRoom().equals(room)) {
```

Comment (dawidjaja) [reply src/main/java/seedu/address/model/Hotel.java]: I'll change this in another branch later.

**Review by dawidjaja — APPROVED** [review]

File: src/main/java/seedu/address/model/Hotel.java [hunk 9]
```diff
@@ -9,12 +9,16 @@
 import javafx.collections.FXCollections;
 import javafx.collections.ObservableList;
 
+import seedu.address.model.hotel.Stay;
```
Comment (dawidjaja) [inline src/main/java/seedu/address/model/Hotel.java]:
```suggestion
```

**Pull request merged**